tests/test_merge_calendars_with_date_recurrence_id.py

```diff
@@ -0,0 +1,23 @@
+from datetime import date, datetime
+from typing import Any
+
+
+def test_merge_calendars_with_date_recurrence_id(merge: Any) -> None:
+    """Test that merging calendars works with both date and datetime recurrence IDs."""
+    merged = merge(["calendar_with_date_recurrence_id.ics"])
+    events = list(merged.walk("VEVENT"))
+
+    # Verify we found events
+    assert len(events) > 0, "Expected to find events in merged calendar"
+
+    # Find events with recurrence IDs
+    recurrence_events = [e for e in events if "RECURRENCE-ID" in e]
+    assert any(recurrence_events), "Expected to find at least one recurring event"
+
+    # Verify recurrence_id can be either date or datetime
+    for event in recurrence_events:
+        recurrence_id = event["RECURRENCE-ID"].dt
+        # Should be either a date or datetime
+        assert isinstance(recurrence_id, (date, datetime)), (
```

Comment:
I think, here one could test if a certain recurrence id is included. The fix could also just delete the event or skip it - would the test then still succeed?

Reply:
Is there any reason we want to convert the recurrence ID to UTC?
If yes, then we should test if the recurrence ID is a datetime object, and only then converted to UTC. Otherwise, leave it as a date.

That's how I think we should fix the issue

Reply:
> Is there any reason we want to convert the recurrence ID to UTC?

I did that for recurring ical events because I wanted to match them (some are without timezone, some with, some use date although they should use datetime). It might not be needed at all for this use case - I would say it is probably worth not converting to UTC and waiting for an issue to arise.

Reply:
Yes I agree
I don't think this is a concern we need to focus on yet

That bring us to the solution of just removing the conversion to UTC, correct?

Reply:
I think yes!

Reply:
less code better code :)